src/wallet/external_signer.cpp

```diff
@@ -0,0 +1,129 @@
+// Copyright (c) 2018-2021 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include <chainparams.h>
+#include <core_io.h>
+#include <psbt.h>
+#include <util/strencodings.h>
+#include <util/system.h>
+#include <wallet/external_signer.h>
+
+ExternalSigner::ExternalSigner(const std::string& command, const std::string& fingerprint, std::string chain, std::string name): m_command(command), m_fingerprint(fingerprint), m_chain(chain), m_name(name) {}
+
+const std::string ExternalSigner::NetworkArg() const
+{
+    if (m_chain == CBaseChainParams::MAIN) {
```

Comment:
I would prefer to introduce a `CBaseChainParams.externalSignerChainName()` instead of an elaborate `if()` here, for this to keep the chain definition data-driven in one place.

Reply:
Alternatively, also out of scope, we could turn `CBaseChainParams` into an enum, so it's easier to use switch statements, and less likely to be forgotten if we add another network type.

Reply:
That'd be another option, but I prefer  to have all this information in one place. It was the point of having chainparams structures in the first place.

Reply:
But good point that they are strings. They also match exactly here:
```
const std::string CBaseChainParams::MAIN = "main";
const std::string CBaseChainParams::TESTNET = "test";
const std::string CBaseChainParams::SIGNET = "signet";
const std::string CBaseChainParams::REGTEST = "regtest";
```
So wouldn't `"" --chain " + m_chain` achieve exactly the same?

Reply:
Good point...